src/overlays/actors/ovl_En_Bom_Chu/z_en_bom_chu.c

```diff
@@ -48,29 +52,440 @@ static ColliderJntSphInit D_809C6D54 = {
         COLSHAPE_JNTSPH,
     },
     1,
-    D_809C6D30,
+    sJntSphElemInit,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(targetMode, 2, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_STOP),
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bom_Chu/EnBomChu_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bom_Chu/EnBomChu_Destroy.s")
+void EnBomChu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static u8 p1StartColor[] = { 250, 0, 0, 250 };
+    static u8 p2StartColor[] = { 200, 0, 0, 130 };
+    static u8 p1EndColor[] = { 150, 0, 0, 100 };
+    static u8 p2EndColor[] = { 100, 0, 0, 50 };
+    EnBomChu* this = THIS;
+    EffectBlureInit1 blureInit;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->colliderElements);
+
+    this->collider.elements[0].dim.worldSphere.radius = this->collider.elements[0].dim.modelSphere.radius;
+
+    for (i = 0; i < 4; i++) {
+        blureInit.p1StartColor[i] = p1StartColor[i];
+        blureInit.p2StartColor[i] = p2StartColor[i];
+        blureInit.p1EndColor[i] = p1EndColor[i];
+        blureInit.p2EndColor[i] = p2EndColor[i];
+    }
+
+    blureInit.elemDuration = 16;
+    blureInit.unkFlag = 0;
+    blureInit.calcMode = 0;
+
+    Effect_Add(globalCtx, &this->blure1Index, EFFECT_BLURE1, 0, 0, &blureInit);
+    Effect_Add(globalCtx, &this->blure2Index, EFFECT_BLURE1, 0, 0, &blureInit);
+
+    this->actor.room = -1;
+    this->timer = 120;
+    this->actionFunc = EnBomChu_WaitForRelease;
+}
+
+void EnBomChu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnBomChu* this = THIS;
+
+    Effect_Delete(globalCtx, this->blure1Index);
+    Effect_Delete(globalCtx, this->blure2Index);
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void EnBomChu_Explode(EnBomChu* this, GlobalContext* globalCtx) {
+    EnBom* bomb;
+    s32 i;
+
+    bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, BOMB_BODY);
+    if (bomb != NULL) {
+        bomb->timer = 0;
+    }
+
+    this->timer = 1;
+    this->actor.speedXZ = 0.0f;
+
+    if (this->actor.yDistToWater > 0.0f) {
+        for (i = 0; i < 40; i++) {
+            EffectSsBubble_Spawn(globalCtx, &this->actor.world.pos, 1.0f, 5.0f, 30.0f, 0.25f);
+        }
+    }
+
+    this->actionFunc = EnBomChu_WaitForKill;
+}
+
+void func_809C5B38(Vec3f* arg0, Vec3f* arg1, Vec3f* dest) {
+    dest->x = (arg0->y * arg1->z) - (arg0->z * arg1->y);
+    dest->y = (arg0->z * arg1->x) - (arg0->x * arg1->z);
+    dest->z = (arg0->x * arg1->y) - (arg0->y * arg1->x);
```

Comment:
```suggestion
void EnBomChu_Cross(Vec3f* a, Vec3f* b, Vec3f* dest) {
    dest->x = (a->y * b->z) - (a->z * b->y);
    dest->y = (a->z * b->x) - (a->x * b->z);
    dest->z = (a->x * b->y) - (a->y * b->x);
```

Reply:
thanks, if you have any other suggestions for the math stuff lmk. either func names or var names